mantle/platform/qemu.go

```diff
@@ -1779,6 +1786,80 @@ func (builder *QemuBuilder) Exec() (*QemuInstance, error) {
 		return nil, err
 	}
 
+	// Process 9p/virtiofs mounts
+	if len(builder.hostMounts) > 0 && builder.UseVirtiofs {
+		if err := builder.ensureTempdir(); err != nil {
+			return nil, err
+		}
+
+		plog.Debug("creating virtiofs helpers")
+
+		// Spawn off a virtiofsd helper per mounted path
+		virtiofsHelpers := make(map[string]exec.Cmd)
+		for i, hostmnt := range builder.hostMounts {
+			// By far the most common failure to spawn virtiofsd will be a typo'd source directory,
+			// so let's synchronously check that ourselves here.
+			if _, err := os.Stat(hostmnt.src); err != nil {
+				return nil, fmt.Errorf("failed to access virtiofs source directory %s", hostmnt.src)
+			}
+			virtiofsChar := fmt.Sprintf("virtiofschar%d", i)
+			virtiofsdSocket := filepath.Join(builder.tempdir, fmt.Sprintf("virtiofsd-%d.sock", i))
+			builder.Append("-chardev", fmt.Sprintf("socket,id=%s,path=%s", virtiofsChar, virtiofsdSocket))
+			builder.Append("-device", fmt.Sprintf("vhost-user-fs-pci,queue-size=1024,chardev=%s,tag=%s", virtiofsChar, hostmnt.dest))
+			plog.Debugf("creating virtiofs helper for %s", hostmnt.src)
+			// TODO: Honor hostmnt.readonly somehow here (add an option to virtiofsd)
+			p := exec.Command("/usr/libexec/virtiofsd", "--sandbox", "none", "--socket-path", virtiofsdSocket, "--shared-dir", ".")
+			p.Dir = hostmnt.src
+			// Quiet the daemon by default
+			p.Env = append(p.Env, "RUST_LOG=ERROR")
+			p.Stderr = os.Stderr
+			p.SysProcAttr = &syscall.SysProcAttr{
+				Pdeathsig: syscall.SIGTERM,
+			}
+			if err := p.Start(); err != nil {
+				return nil, fmt.Errorf("failed to start virtiofsd")
+			}
+			virtiofsHelpers[virtiofsdSocket] = p
+		}
+		// Loop waiting for the sockets to appear
+		delay := time.Millisecond * 250
+		repetitions := time.Minute / delay
+		for i := 0; i < int(repetitions) && len(virtiofsHelpers) > 0; i++ {
+			// Delay between subsequent loops
+			if i > 0 {
+				time.Sleep(delay)
+			}
+			found := []string{}
+			for sockpath := range virtiofsHelpers {
+				if _, err := os.Stat(sockpath); err == nil {
+					found = append(found, sockpath)
+				}
+			}
+			for _, sockpath := range found {
+				delete(virtiofsHelpers, sockpath)
```

Comment:
Shouldn't we be adding to `inst.helpers` here before deleting?

Reply:
Yes, good catch!  (Though in practice actually it doesn't really matter since the only role of that is to call `Kill()` on terminate but we already use pdeathsig to ensure the virtiofsd instances die if qemu dies)